Xamarin.Forms.Maps.Android/MapRenderer.cs

```diff
@@ -10,311 +11,358 @@
+			}
+			else if (changed)
+			{
+				UpdateVisibleRegion(NativeMap.CameraPosition.Target);
 			}
 		}
 
-		public void OnCameraChange (CameraPosition pos)
-		{
-			UpdateVisibleRegion (pos.Target);
-		}
-
-		void UpdateVisibleRegion (LatLng pos)
-		{
-			var map = NativeMap;
-			if (map == null)
-				return;
-			var projection = map.Projection;
-			var width = Control.Width;
-			var height = Control.Height;
-			var ul = projection.FromScreenLocation (new global::Android.Graphics.Point (0, 0));
-			var ur = projection.FromScreenLocation (new global::Android.Graphics.Point (width, 0));
-			var ll = projection.FromScreenLocation (new global::Android.Graphics.Point (0, height));
-			var lr = projection.FromScreenLocation (new global::Android.Graphics.Point (width, height));
-			var dlat = Math.Max (Math.Abs (ul.Latitude - lr.Latitude), Math.Abs (ur.Latitude - ll.Latitude));
-			var dlong = Math.Max (Math.Abs (ul.Longitude - lr.Longitude), Math.Abs (ur.Longitude - ll.Longitude));
-			((Map)Element).VisibleRegion = new MapSpan (
-					new Position (
-						pos.Latitude,
-						pos.Longitude
-					),
-				dlat,
-				dlong
-			);
-		}
-
-		void AddPins (IList pins)
+		void AddPins(IList pins)
 		{
-			var map = NativeMap;
+			GoogleMap map = NativeMap;
 			if (map == null)
+			{
 				return;
+			}
 
 			if (_markers == null)
-				_markers = new List<Marker> ();
+			{
+				_markers = new List<Marker>();
+			}
 
-			_markers.AddRange( pins.Cast<Pin>().Select (p => {
-				var pin = (Pin)p;
-				var opts = new MarkerOptions ();
-				opts.SetPosition (new LatLng (pin.Position.Latitude, pin.Position.Longitude));
-				opts.SetTitle (pin.Label);
-				opts.SetSnippet (pin.Address);
-				var marker = map.AddMarker (opts);
+			_markers.AddRange(pins.Cast<Pin>().Select(p =>
+			{
+				Pin pin = p;
+				var opts = new MarkerOptions();
+				opts.SetPosition(new LatLng(pin.Position.Latitude, pin.Position.Longitude));
+				opts.SetTitle(pin.Label);
+				opts.SetSnippet(pin.Address);
+				Marker marker = map.AddMarker(opts);
 
 				// associate pin with marker for later lookup in event handlers
 				pin.Id = marker.Id;
 				return marker;
 			}));
 		}
 
-		void RemovePins (IList pins)
-		{
-			var map = NativeMap;
-			if (map == null)
-				return;
-			if (_markers == null)
-				return;
-			
-			foreach (Pin p in pins) {
-				var marker = _markers.FirstOrDefault (m => (object)m.Id == p.Id);
-				if (marker == null)
-					continue;
-				marker.Remove ();
-				_markers.Remove (marker);
-			}
-		}
-
-		void MapOnMarkerClick (object sender, GoogleMap.InfoWindowClickEventArgs eventArgs)
+		void MapOnMarkerClick(object sender, GoogleMap.InfoWindowClickEventArgs eventArgs)
 		{
 			// clicked marker
-			var marker = eventArgs.Marker;
+			Marker marker = eventArgs.Marker;
 
 			// lookup pin
 			Pin targetPin = null;
-			for (var i = 0; i < Map.Pins.Count; i++) {
-				var pin = Map.Pins[i];
+			for (var i = 0; i < Map.Pins.Count; i++)
+			{
+				Pin pin = Map.Pins[i];
```

Comment:
Trying to follow [rule 10](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md), which ReSharper has a setting for. But ReSharper's decision here is odd; it's pretty obvious what the type is. I'm going to change that one back.